rust/main/agents/relayer/src/msg/op_queue.rs

```diff
@@ -72,27 +72,66 @@ impl OpQueue {
         // The other consideration is whether to put the channel receiver in the OpQueue or in a dedicated task
         // that also holds an Arc to the Mutex. For simplicity, we'll put it in the OpQueue for now.
         let mut message_retry_requests = vec![];
-        while let Ok(message_id) = self.retry_rx.lock().await.try_recv() {
-            message_retry_requests.push(message_id);
+
+        // we only need to lock self.retry_receiver once
+        {
+            let mut retry_receiver = self.retry_receiver.lock().await;
```

Comment:
not sure if this really matters but I think this way, we only lock once instead of on each iteration